internal/zinc-core/src/main/scala/sbt/internal/inc/APIDiff.scala

```diff
@@ -57,29 +30,211 @@ private[inc] class APIDiff {
                       contextSize: Int): String = {
     val api1Str = DefaultShowAPI(api1.classApi) + "\n" + DefaultShowAPI(api1.objectApi)
     val api2Str = DefaultShowAPI(api2.classApi) + "\n" + DefaultShowAPI(api2.objectApi)
-    generateApiDiff(fileName, api1Str, api2Str, contextSize)
+    DiffUtil.mkColoredCodeDiff(api1Str, api2Str, printDiffDel = true)
   }
 
-  private def generateApiDiff(fileName: String, f1: String, f2: String, contextSize: Int): String = {
-    assert((diffMethod != null) && (generateUnifiedDiffMethod != null),
-           "APIDiff isn't properly initialized.")
-    import scala.collection.JavaConverters._
-    def asJavaList[T](it: Iterator[T]): java.util.List[T] = it.toSeq.asJava
-    val f1Lines = asJavaList(f1.lines)
-    val f2Lines = asJavaList(f2.lines)
-    //val diff = DiffUtils.diff(f1Lines, f2Lines)
-    val diff /*: Patch*/ = diffMethod.invoke(null, f1Lines, f2Lines)
-    val unifiedPatch: JList[String] = generateUnifiedDiffMethod
-      .invoke(null, fileName, fileName, f1Lines, diff, (contextSize: java.lang.Integer))
-      .asInstanceOf[JList[String]]
-    unifiedPatch.asScala.mkString("\n")
-  }
+  /**
+   * This class was directly copied from Dotty [[https://github.com/lampepfl/dotty/blob/0.1.2-RC1/compiler/src/dotty/tools/dotc/util/DiffUtil.scala]]
+    *
+    * Copyright (c) 2014, EPFL
+    *
+    * All rights reserved.
+    * Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+    * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+    * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+    * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+    * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+   */
```

Comment:
Not a problem at all :) I just assumed no one will really read it anyway. Fixed now.